homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -254,6 +257,19 @@ def register_services(hass: HomeAssistant) -> None:
             ),
         }
     )
+    service_send_transp_cmd_schema = vol.Schema(
+        {
+            vol.Required(ATTR_GW_ID): vol.All(
+                cv.string, vol.In(hass.data[DATA_OPENTHERM_GW][DATA_GATEWAYS])
+            ),
+            vol.Required(ATTR_TRANSP_CMD): vol.All(
+                cv.string, vol.Length(min=2, max=2), vol.Coerce(str.upper)
+            ),
+            vol.Required(ATTR_TRANSP_ARG): vol.All(
+                cv.string, vol.Length(min=1, max=12)
```

Comment:
Where does the max length come from?

Reply:
The gateway has a 16 byte RX buffer.  (https://github.com/hvxl/otgw/blob/master/gateway.asm#L571)

Subtract the command, equal sign and terminator from this buffer and you will have room for 12 argument bytes.